src/components/CreateSmartList/index.js

```diff
@@ -333,96 +290,103 @@ const CreateSmartList = () => {
     })
   }
 
-  const sortedGroceriesByDaysAverage = arrayOfItemsByAverage[0].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
-  )
+  const sortdaysAverage = (array) =>
+    array.sort((a, b) => b.daysAverage - a.daysAverage)
 
-  const sortedProductsByDaysAverage = arrayOfItemsByAverage[1].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
+  const sortedGroceriesByDaysAverage = sortdaysAverage(arrayOfItemsByAverage[0])
+  const sortedProductsByDaysAverage = sortdaysAverage(arrayOfItemsByAverage[1])
+  const sortedDishesByDaysAverage = sortdaysAverage(arrayOfItemsByAverage[2])
+  const sortedSavedListsByDaysAverage = sortdaysAverage(
+    arrayOfItemsByAverage[3],
   )
 
-  const sortedDishesByDaysAverage = arrayOfItemsByAverage[2].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
-  )
+  const sortdaysValue = (array) => array.sort((a, b) => b.value - a.value)
+
+  const sortedGroceriesByValueFromHistory = sortdaysValue(workingArraysList[0])
+  const sortedProductsByValueFromHistory = sortdaysValue(workingArraysList[1])
+  const sortedDishesByValueFromHistory = sortdaysValue(workingArraysList[2])
+  const sortedSavedListByValueFromHistory = sortdaysValue(workingArraysList[3])
+
+  const propsItemsList = {
+    Groceries: sortByNumber
+      ? sortedGroceriesByValueFromHistory
+      : sortedGroceriesByDaysAverage,
+    Products: sortByNumber
+      ? sortedProductsByValueFromHistory
+      : sortedProductsByDaysAverage,
+    Dishes: sortByNumber
+      ? sortedDishesByValueFromHistory
+      : sortedDishesByDaysAverage,
+    SavedList: sortByNumber
+      ? sortedSavedListByValueFromHistory
+      : sortedSavedListsByDaysAverage,
+  }
 
-  const sortedSavedListsByDaysAverage = arrayOfItemsByAverage[3].sort(
-    (a, b) => b.daysAverage - a.daysAverage,
-  )
-
-  const sortedGroceriesByValueFromHistory = workingArraysList[0].sort(
-    (a, b) => b.value - a.value,
-  )
-  const sortedProductsByValueFromHistory = workingArraysList[1].sort(
-    (a, b) => b.value - a.value,
-  )
-  const sortedDishesByValueFromHistory = workingArraysList[2].sort(
-    (a, b) => b.value - a.value,
-  )
-  const sortedSavedListByValueFromHistory = workingArraysList[3].sort(
-    (a, b) => b.value - a.value,
-  )
+  const getEditModeListProps = (
+    data,
+    parentsTitle,
+    itemsList,
+    filtredCategories,
+    postItemToBasket,
+    putBasketItem,
+    updateData,
+  ) => ({
+    BasketitemsList: data,
+    parentsTitle,
+    itemsList: itemsList[parentsTitle],
+    filtredCategories,
+    variant: 'shop',
+    parentCategories: parentCategories,
```

Comment:
helper super, ta linijka wystarczy `parentCategories,` chyba Ci Uciekło

Reply:
racja, masz oko sokoła :) ale to dobrze xd

Reply:
poprawione